### ISSUE TITLE
Fail to apply a string replace function to series

### ISSUE THREAD

**Issue by zlyin** [jmcarpenter2/swifter#178]
Hello there,
  I'm new to this useful tool and I just came across a potential bug. The problem is described as follows.
  I have got a csv with bad image names, where ".avi" was wrongly put there. For example, 
   ```
  image
  004204020000000_1860.jpg
  004204020000000_1880.jpg
  004204020000000_1900.jpg
  004204020000000_1920.jpg
  004204020000000_1940.jpg
  004204020000000_1960.jpg
  004204020000000_1980.jpg
   ```
 I attempted to replace ".avi_", with "_" to receive clean filenames. But `swifter.apply` failed to do the replacement work. Here is my experiment versus the original pandas apply function.

```
import pandas as pd
import swifter

df = pd.read_csv("./debug_swifter.csv")
print(f"--> check orig data=\n{df.tail()}")

df1 = df.copy()
df2 = df.copy()

fix_func = lambda x : x.replace(".avi_", "_")

# way 1 --> use pandas orig apply
df1["image"] = df1.image.apply(fix_func)
print(f"--> after pandas orig apply =\n{df1.tail()}")

# way 2 --> use col.swifter.apply
df2["image"] = df2.image.swifter.apply(fix_func)
print(f"--> after col.swifter.apply =\n{df2.tail()}")
```
And the outputs are as follows.
```
--> check orig data=
                             image
30935   023531030000000.avi_93.jpg
30936  023531030000000.avi_935.jpg
30937  023531030000000.avi_938.jpg
30938  023531030000000.avi_941.jpg
30939  023531030000000.avi_946.jpg
--> after pandas orig apply =
                         image
30935   023531030000000_93.jpg
30936  023531030000000_935.jpg
30937  023531030000000_938.jpg
30938  023531030000000_941.jpg
30939  023531030000000_946.jpg
--> after col.swifter.apply =
                             image
30935   023531030000000.avi_93.jpg
30936  023531030000000.avi_935.jpg
30937  023531030000000.avi_938.jpg
30938  023531030000000.avi_941.jpg
30939  023531030000000.avi_946.jpg
```

I have also uploaded my debug csv, please have a look and see if this is a bug.

Thank you very much.

[debug_swifter.csv](https://github.com/jmcarpenter2/swifter/files/8469156/debug_swifter.csv)

**Comment by jmcarpenter2** [comment]
Thank you SO MUCH @zlyin for the very clear example.

I found the bug.... it was because within swifter it tests the first 10-20 rows to check if the vectorized apply is the same output as the pandas apply and then it just returns vectorized solution.

However, in this data set, the first 10-20 rows don't have `avi_` in the str field.

So it thought the vectorized solution was correct when it wasnt!

I am going to put up a pull request to select random 10-20 rows to check for vectorized apply rather than just the first 10-20. This will increase the likelihood that the vectorized check actually verifies if its the thing to do (fastest correct solution)

Example with the fix:
![Screen Shot 2022-04-12 at 10 31 19 AM](https://user-images.githubusercontent.com/29585347/163020406-c1c02b9e-e6fd-475a-8ec5-f0f4eaa61d88.png)

**Comment by jmcarpenter2** [comment]
Closed by #179 
Please use release `1.1.3` now :D

**Comment by zlyin** [comment]
Hello Jason, 
  Great thanks for your fast reply and fixup! I'm more than happy to provide my help. 
  Another tips I could provide is about your solution. 
  
> I am going to put up a pull request to select random rows (more than 10-20 as well) to check for vectorized apply rather than just the first 10-20.
  
  The first thing is about the sampling amount. I play with huge dataframes a lot, and I think 5% is a good threshold to check the pattern. Maybe you can design a step function that assigns different sampling numbers w.r.t various amounts of data. 
  The second thing that concerns you might be the sampling method. Instead of a method in the fashion of Gaussian distribution, some unevenly distribution could be used. This helps to reveal the potential bug and ensure correct output.
